### ISSUE TITLE
Doesn't work for rust wasm worker

### ISSUE THREAD

**Issue by alank976** [cloudflare/wrangler-action#16]
It failed when it tried to install wasm-pack during `wrangler publish`
```
Wrangler has been installed!
+ @cloudflare/wrangler@1.6.0
added 27 packages from 13 contributors in 1.957s
⬇️ Installing wasm-pack...
 Compiling your project to WebAssembly...
Error: No such file or directory (os error 2)
Error: tried running command:
/github/workspace/.cache/.wrangler/wasm-pack-c6ddc20cc0d48874/wasm-pack build --target no-modules
```
I managed to reproduce it in my forked branch here https://github.com/alank976/wrangler-action/commit/2d7e8ad358457eea95676cad89f524edf213ab6e:
1. `wrangler generate test-rust-wasm https://github.com/cloudflare/rustwasm-worker-template` and tweak a bit `zone_id`, `route`, and `account_id` in wrangler.toml in order to suppress complaints about authentication
2. use docker-compose to reproduce the github action flow and mount the generated wasm project as `INPUT_WORKINGDIRECTORY`

You should be able to reproduce the same by running `docker-compose up`

I will probably try to fix by explicitly installing `wasm-pack` before `wrangler publish` and hopefully I can create a PR tomorrow for this :)

**Comment by alank976** [comment]
After diving into it, basically this action doesn't support rust-wasm worker at the moment as its [`wasm-pack` requires Rust runtime](https://github.com/rustwasm/wasm-pack#-prerequisities).

There are a few solutions in my mind:
1. use different dockerfile conditionally based on `type` in `wrangler.toml`
1. use a dockerfile which has both node and rust installed
1. make a docker image available in dockerhub which is preinstalled wrangler, wasm-pack, and rust. Its version should be in sync with wrangler CLI version

**Comment by signalnerve** [comment]
Ah, fascinating! We may want to revisit how we configure the Docker part of this to allow WASM deployments – you're totally right that this is JS-focused at the moment. cc @cloudflare/workers-devexp

**Comment by ashleygwilliams** [comment]
wasm-pack is distributed as a prebuilt binary- so there's no "runtime" required... however wasm-pack does call the rust compiler to compile your rust to wasm, so the rust compiler does need to be installed.

happy to help figure this out and get out a new version!

**Comment by alank976** [comment]
@signalnerve @ashleygwilliams Thank you for clarification. I am trying to make a PR with option 3 in my mind. Will tag you guys for code review :)

**Comment by hmatt1** [comment]
Has there been any progress on this? It would be great to see the GitHub Action support rust workers as well!

For reference, the [rust toolchain action](https://github.com/actions-rs/toolchain) lets you specify whether you're using the nightly build, stable, or a specific version. It would be nice if the Cloudflare Wrangler Action would also let you specify which rust versions you want to use.

In PR https://github.com/cloudflare/wrangler-action/pull/18, it looks like the proposed solution is to build rust into the base Docker image. I think if we went this route, it would be limited to that specific version of rust. We really shouldn't have to rebuild the base image all the time for anyone using nightly builds, or whenever new versions of rust come out.

**Comment by haasal** [comment]
Tl;dr: IMO this is an easy fix, I have already implemented it in a fork. What's holding us up? If there are any issues with this I am of course open for discussion. 

I don't quite understand the issue here and all the closed PRs on this. This seems like an increasingly easy fix:

First option:

Use the official Rust:alpine Docker image.
This requires adding two commands for adding the wasm toolchain and three other dependencies (node, npm, bash and some library used for compiling to wasm in rust).

Second option:

Leave it at the node:16 image but install rust (ideally without the unnecessary features like rustfmt etc.) inside the entrypoint script or inside the Dockerfile. 

Also this is a bit of-topic but why aren't we replacing wrangler publish with wrangler deploy inside the entrypoint script. publish is deprecated and apparently almost the same as deploy.
